crates/matrix-sdk-ui/src/timeline/controller/state.rs

```diff
@@ -199,13 +199,15 @@ impl TimelineState {
         thread_root: &Option<OwnedEventId>,
     ) -> bool {
         match &timeline_focus {
-            TimelineFocusKind::Live => thread_root.is_none(),
+            TimelineFocusKind::Live { hide_threaded_events } => {
+                thread_root.is_none() || thread_root.is_some() && !hide_threaded_events
```

Comment:
In the right hand side, we can remove `thread_root.is_some()`, since it's implied from the negation of the LHS. Boole's logic ftw!